src/components/ContentWidgets/ImagesInRow/ImagesInRowEditor.tsx

```diff
@@ -0,0 +1,201 @@
+import React, { useState } from 'react';
+import { WidgetEditorProps } from '../../ContentMapping/ContentMapping';
+import './ImagesInRow.css';
+import { Button, TextField } from '@material-ui/core';
+
+type imgRowItem = {
+    imgLink: string;
+    imgCaption: string;
+    imgAltTag: string;
+    externalLink: string;
+}
+
+/**
+ * Diplays an editor to create/edit an ImageInRow widget
+ *
+ * Last Modified
+ * Nitesh Chetry
+ * July 19, 2019
+ */
+export const ImagesInRowEditor: React.FC<WidgetEditorProps> = ({
+    editedContent,
+    setEditedContentOnChange
+}) => {
+    const [imgLinkText, changeImgLinkText] = useState("");
+    const [imgCaptionText, changeCaptionText] = useState("");
+    const [imgAltTagText, changeImgAltTagText] = useState("");
+    const [externalLink, changeExternalLink] = useState("");
+    const [emptyImgLinkSubmitted, switchEmptyImgLinkSubmitted] = useState(false);
+    const [emptyCaptionAndAltTagSubmitted, switchEmptyCaptionAndAltTagSubmitted] = useState(false);
+    let pictureCaptionPairs: imgRowItem[] = editedContent.imagesInRow_items || [];
+
+    const addImage = (imgLinkText: string, imgCaptionText: string, imgAltTagText: string,
+        externalLink: string) => {
+        if (imgLinkText !== "" && (imgCaptionText !== "" || imgAltTagText !== "")) {
+            pictureCaptionPairs.push({
+                imgLink: imgLinkText,
+                imgCaption: imgCaptionText,
+                imgAltTag: imgAltTagText,
+                externalLink: externalLink
+            });
+            setEditedContentOnChange("imagesInRow_items", pictureCaptionPairs);
+            changeImgLinkText("");
+            changeCaptionText("");
+            changeImgAltTagText("");
+            changeExternalLink("");
+        }
+    }
+
+    if (!pictureCaptionPairs) {
+        return <></>;
+    }
+    return <>
+        <h3> Add New Image </h3>
+        {emptyImgLinkSubmitted ?
+            <>
+                <h4 className="error-message"> Image Link must not be empty</h4>
+                <TextField
```

Comment:
Can you add some inline comments to just make it easy to see what text fields these are for? Ideally we would have made it easier to read or more componentized but I honestly can't see how to really right now.